### DISCUSSION TITLE
Invalid API Key

### DISCUSSION THREAD

**Discussion by josh99smith** [Budibase/budibase#11533] [Help] [answered]
Hello,

I am trying to create some custom querries to my own internal budibase base. However, the editor keeps giving me the following response.

{
  "message": "Invalid API key - may need re-generated, or user doesn't exist",
  "status": 403
}

I tried to regenerate my API key mulitple times, but I still get this error. Anyone know what could be causing this?

**Comment by melohagan** [comment]
Hey @josh99smith 

I have a vague memory of getting this error myself, but it turning out to be slightly misleading. Make sure everything else about your query is correct, including the right headers and body content.

**Comment by AlejandroFNadal** [comment, marked as answer]
I had to use the Internal Budibase API Key and not the one that is given by Budibase. With internal, I mean the one that is placed in the .env file when you run the budibase as a docker compose. The one that is given in the front also required a CSRF Token to be used.

**Reply by melohagan** [reply]
Ah that would be correct, thanks for the answer! I had assumed the public API was being used here, but for internal queries against the DB then you would need the key from the `.env` file.
